src/sparseml/pytorch/datasets/classification/_ffcv/__ffcv.py

```diff
@@ -0,0 +1,234 @@
+            },
+            num_workers=num_workers,
+        )
+
+        LOGGER.info(f"Writing dataset to {self.write_path}")
+        writer.from_indexed_dataset(self.dataset)
+
+    @contextmanager
+    def switched_off_transform(self):
+        """
+        Context manager to switch off the transforms in original dataset,
+        and then switch it back on after the context is exited.
+        """
+        _old_transform = None
+        _old_target_transform = None
+
+        try:
+            # Switch off the transforms
+            if hasattr(self.dataset, "transform"):
+                _old_transform = self.dataset.transform
+                self.dataset.transform = None
+                LOGGER.info("Transforms switched off for original dataset.")
+
+            if hasattr(self.dataset, "target_transform"):
+                _old_transform = self.dataset.target_transform
+                self.dataset.transform = None
+                LOGGER.info("Target Transforms switched off for original dataset.")
+
+            # Give back control
+            yield
+
+        finally:
+            # restore the transforms
+            if _old_transform is not None and hasattr(self.dataset, "transform"):
+                self.dataset.transform = _old_transform
+                LOGGER.info("Transforms switched on for original dataset.")
+
+            if _old_target_transform is not None and hasattr(
+                self.dataset, "target_transform"
+            ):
+                self.dataset.target_transform = _old_target_transform
+                LOGGER.info("Target Transforms switched on for original dataset.")
+
+    def get_loader(
+        self,
+        num_workers: int = 16,
+        batch_size: int = 32,
+        distributed: int = 0,
+        in_memory: int = 0,
+        validation: bool = False,
+    ):
+        """
+        Initialize the data loader.
+
+        :param num_workers: Number of workers to use for the data loader.
+        :param batch_size: Batch size for the data loader.
+        :param distributed: 1/0 Whether to use distributed data loading.
+        :param in_memory: 1/0 Does the dataset fit in memory.
+        """
+
+        _init_msg = "Initializing data loader."
+        if validation:
+            _init_msg = "Initializing validation data loader."
+        LOGGER.info(_init_msg)
+
+        resolution = (self._image_size, self._image_size)
+        decoder = (
+            CenterCropRGBImageDecoder(
+                resolution,
+                ratio=self.DEFAULT_CROP_RATIO,
+            )
+            if validation
+            else RandomResizedCropRGBImageDecoder(resolution)
+        )
+        image_pipeline: List[Operation] = [
+            decoder,
+            RandomHorizontalFlip(),
+            ToTensor(),
+            ToDevice(torch.device(self._device), non_blocking=True),
+            ToTorchImage(),
+            NormalizeImage(self.IMAGENET_MEAN, self.IMAGENET_STD, np.float32),
```

Comment:
might want to have an overridable property for mean/std since some datasets might want to use a different one

Reply:
Let's sync on this offline